eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Genesis.hs

```diff
@@ -145,49 +204,43 @@ instance FromCBOR AlonzoGenesis where
           <! From
 
 instance ToCBOR AlonzoGenesis where
-  toCBOR
-    AlonzoGenesis
-      { agCoinsPerUTxOWord
-      , agCostModels
-      , agPrices
-      , agMaxTxExUnits
-      , agMaxBlockExUnits
-      , agMaxValSize
-      , agCollateralPercentage
-      , agMaxCollateralInputs
-      } =
-      toEraCBOR @AlonzoEra
-        . encode
-        $ Rec AlonzoGenesis
-          !> To agCoinsPerUTxOWord
-          !> To agCostModels
-          !> To agPrices
-          !> To agMaxTxExUnits
-          !> To agMaxBlockExUnits
-          !> To agMaxValSize
-          !> To agCollateralPercentage
-          !> To agMaxCollateralInputs
+  toCBOR x@(AlonzoGenesis _ _ _ _ _ _ _ _ _) =
+    let AlonzoGenesis {..} = x
+     in toEraCBOR @AlonzoEra . encode $
+          Rec AlonzoGenesis
+            !> To agCoinsPerUTxOWord
+            !> E encodeCostModel agCostModels
+            !> To agPrices
+            !> To agMaxTxExUnits
+            !> To agMaxBlockExUnits
+            !> To agMaxValSize
+            !> To agCollateralPercentage
+            !> To agMaxCollateralInputs
+            !> To agExtraConfig
 
 instance FromJSON AlonzoGenesis where
   parseJSON = Aeson.withObject "Alonzo Genesis" $ \o -> do
     agCoinsPerUTxOWord <- o .: "lovelacePerUTxOWord"
-    agCostModels <- parseCostModels False =<< o .: "costModels"
+    cms <- parseCostModels False =<< o .: "costModels"
     agPrices <- o .: "executionPrices"
     agMaxTxExUnits <- o .: "maxTxExUnits"
     agMaxBlockExUnits <- o .: "maxBlockExUnits"
     agMaxValSize <- o .: "maxValueSize"
     agCollateralPercentage <- o .: "collateralPercentage"
     agMaxCollateralInputs <- o .: "maxCollateralInputs"
+    agExtraConfig <- o .:? "extraConfig"
+    let agCostModels = costModelsValid cms ! PlutusV1
     return AlonzoGenesis {..}
 
 instance ToKeyValuePairs AlonzoGenesis where
   toKeyValuePairs ag =
     [ "lovelacePerUTxOWord" .= agCoinsPerUTxOWord ag
-    , "costModels" .= agCostModels ag
+    , "costModels" .= mkCostModels (Map.singleton PlutusV1 $ agCostModels ag)
     , "executionPrices" .= agPrices ag
     , "maxTxExUnits" .= agMaxTxExUnits ag
     , "maxBlockExUnits" .= agMaxBlockExUnits ag
     , "maxValueSize" .= agMaxValSize ag
     , "collateralPercentage" .= agCollateralPercentage ag
     , "maxCollateralInputs" .= agMaxCollateralInputs ag
     ]
+      ++ ["extraConfig" .= agExtraConfig ag | isJust (agExtraConfig ag)]
```

Comment:
An alternative way to do the same thing. Feel free to pick the one you like better.
```suggestion
      ++ ["extraConfig" .= extraConfig | Just extraConfig <- [agExtraConfig ag]]
```